.common-ci.yml

```diff
@@ -0,0 +1,211 @@
+    - if: $CI_COMMIT_TAG && $CI_COMMIT_TAG != ""
+
+workflow:
+  rules:
+    # We trigger the pipeline on a merge request
+    - if: $CI_PIPELINE_SOURCE == 'merge_request_event'
+    # We then add all the regular triggers
+    - !reference [.pipeline-trigger-rules, rules]
+
+# The main or manual job is used to filter out distributions or architectures that are not required on
+# every build.
+.main-or-manual:
+  rules:
+    - !reference [.pipeline-trigger-rules, rules]
+    - if: $CI_PIPELINE_SOURCE == "schedule"
+      when: manual
+
+# The trigger-pipeline job adds a manualy triggered job to the pipeline on merge requests.
+trigger-pipeline:
+  stage: trigger
+  script:
+    - echo "starting pipeline"
+  rules:
+    - !reference [.main-or-manual, rules]
+    - if: $CI_PIPELINE_SOURCE == "merge_request_event"
+      when: manual
+      allow_failure: false
+    - when: always
+
+# The .dist- dummy steps set the DIST variable for the targeted distribution.
+.dist-distroless:
+  variables:
+    DIST: "distroless"
+
+# Define the platform targets
+.platform-amd64:
+  variables:
+    PLATFORM: linux/amd64
+
+.platform-arm64:
+  variables:
+    PLATFORM: linux/arm64
+
+# Make buildx available as a docker CLI plugin
+.buildx-setup:
+  before_script:
+    -  export BUILDX_VERSION=v0.6.3
+    -  apk add --no-cache curl
+    -  mkdir -p ~/.docker/cli-plugins
+    -  curl -sSLo ~/.docker/cli-plugins/docker-buildx "https://github.com/docker/buildx/releases/download/${BUILDX_VERSION}/buildx-${BUILDX_VERSION}.linux-amd64"
+    -  chmod a+x ~/.docker/cli-plugins/docker-buildx
+
+    -  docker buildx create --use --platform=linux/amd64,linux/arm64
+
+    -  '[[ -n "${SKIP_QEMU_SETUP}" ]] || docker run --rm --privileged multiarch/qemu-user-static --reset -p yes'
+
+# The .scan step forms the base of the image scan operation performed before releasing
+# images. We implement a .scan-base job here to allow for the variable checks to be
+# exercised before internal releases.
+.scan-base:
+  stage: scan
+  variables:
+    IMAGE: "${CI_REGISTRY_IMAGE}/k8s-nim-operator:${CI_COMMIT_SHORT_SHA}"
+    IMAGE_ARCHIVE: "k8s-nim-operator.tar"
+  except:
+    variables:
+      - $SKIP_SCANS && $SKIP_SCANS == "yes"
+  before_script:
+    - docker login -u "${CI_REGISTRY_USER}" -p "${CI_REGISTRY_PASSWORD}" "${CI_REGISTRY}"
+    - docker pull --platform="${PLATFORM}" "${IMAGE}"
+    - docker save "${IMAGE}" -o "${IMAGE_ARCHIVE}"
+  script:
+    - >
+      echo "Skipping scan of ${IMAGE}"
+
+.scan:
+  extends:
+    - .scan-base
+
+# Define the scan targets
+scan-distroless-amd64:
+  extends:
+    - .scan
+    - .dist-distroless
+    - .platform-amd64
+  needs:
+    - image-distroless
+
+scan-distroless-arm64:
+  extends:
+    - .scan
+    - .dist-distroless
+    - .platform-arm64
+  needs:
+    - image-distroless
+    - scan-distroless-amd64
+
+# Download the regctl binary for use in the release steps
+.regctl-setup:
+  before_script:
+    - export REGCTL_VERSION=v0.3.10
```

Comment:
nit: similar to my comment on buildx, can we upgrade the regctl version to at least v0.7.1? https://github.com/NVIDIA/gpu-operator/blob/main/.common-ci.yml#L153